### PR TITLE
Allow Article Schema on Pages and CPTs

### PR DESCRIPTION
**Pull request by ydlr**
This allows article schema to be added to all pages and post types, not just blog posts.

It uses `$current_screen->base` instead of `$current_screen->post_type` to determine if the current screen is a single post edit/add screen.

### CODE REVIEW

**Comment by sybrew** [comment]
Thank you so much for the PR! You're the first to make such effort, and I'm very grateful 💗

I have a very strict coding and usage policy for TSFEM; so, I may come off harsh, but it's purely syntaxical.

Let me walk quickly through this:

------------------

***There's already a function available for detecting all "post-type" posts in the admin, using the aforementioned `base`.***

[From query.class.php](https://github.com/sybrew/the-seo-framework/blob/668cbccc336dab1ab6c5626d8eb50ed7350605e8/inc/classes/query.class.php#L854-L869):
```php
	/**
	 * Determines if the page is singular within the admin screen.
	 *
	 * @since 2.5.2
	 * @global object $current_screen
	 *
	 * @return bool Post Type is singular
	 */
	public function is_singular_admin() {
		global $current_screen;

		if ( isset( $current_screen->base ) && ( 'edit' === $current_screen->base || 'post' === $current_screen->base ) )
			return true;

		return false;
	}
```

------------------

***There's a minor bug in it.***

The [the_seo_framework()->is_single() check](https://github.com/sybrew/The-SEO-Framework-Extension-Manager/pull/3/commits/54abb75aa8267554dcdcf223582628d6a3e50a18#diff-7e7c34f640eb6743b57153674bc2aee2) still prevents the output on pages. Although, it does allow it on non-hierarchical CPT.

It should use `is_singular()` in the case you describe... however...

------------------

***I'm not in favor for this change.***

In short, I'd rather have no markup than wrong markup.

From https://learn.wordpress.com/get-published/

<blockquote>Posts are what you think of when you think of a blog; they’re dated, and appear in reverse chronological order. They move down your blog as you publish new ones.

[...]
Pages are different — they’re handy for timeless content, like an About or Contact Me pages.</blockquote>

The Articles markup is not for timeless content.
I dove deeply into this subject before making this decision. It's described here:
https://theseoframework.com/extensions/articles/#usage/types

As for CPT, it's difficult to detect whether one's meant to be an Article or not.

Could you clarify your reasoning? I could make adjustments based on this.

**Comment by ydlr** [comment]
I guess I didn't think nearly enough about the general uses of this before hand. In my case, I have a large number of pages ( or sometimes CPTs ) with information on specific topics. The content is similar to what a blog post might be used for, I just happen to have them separate from blog.

Maybe a filter that accepts a list of post types would be better?

**Comment by sybrew** [comment]
Hi @ydlr,

A filter sounds great. How it's going to work within the plugin is still uncertain. 
I need to look for reusable and stable patterns for both the front-end as back-end.
